src/fifolock.jl

```diff
@@ -0,0 +1,143 @@
+@static if VERSION >= v"1.10"
+
+const LOCKED_BIT = 0b01
+
+"""
+    FIFOLock()
+
+Like Base.ReentrantLock, but with strict FIFO ordering.
+"""
+mutable struct FIFOLock <: AbstractLock
```

Comment:
I think you also need to specify in the docstring that this is also Reentrant. Maybe it should be called `FIFOReentrantLock`? Or `ReentrantFIFOLock`?

Reply:
Or maybe just putting an explicit note in the docstring is enough.

Reply:
Updated the docstring.